common/djangoapps/student/admin.py

```diff
@@ -183,6 +178,43 @@ class CourseEnrollmentAdmin(admin.ModelAdmin):
     def queryset(self, request):
         return super(CourseEnrollmentAdmin, self).queryset(request).select_related('user')
 
+    def has_permission(self, request, method):
+        """
+        Returns True if the given method is allowed.
+
+        Access to these admin views is restricted because it makes DB quries that impact performance enough to cause a
+        site outage, cf https://openedx.atlassian.net/browse/OPS-2943
+
+        Enable these views using the waffle switch: `student.coursenrollment.admin`
+        """
+        if waffle.switch_is_active('student.coursenrollment.admin'):
+            return getattr(super(CourseEnrollmentAdmin, self), method)(request)
+        return False
```

Comment:
I found this more readable:

```python
has_permission = getattr(super(CourseEnrollmentAdmin, self), method)(request)
return waffle.switch_is_active('student.coursenrollment.admin') and has_permission
```

or it could be shortened to a one-liner by removing the variable.

Reply:
Wow, really?  I find that harder to read than what I wrote..  also easier to show branch test coverage with an explicit if.

Will leave it up to upstream :)